scripts/setVersion.js

```diff
@@ -32,8 +32,13 @@ const updateMgtDependencyVersion = (packages, version) => {
     console.log(`updating package ${package} with version ${version}`);
     const data = fs.readFileSync(package, 'utf8');
 
-    var result = data.replace(/"(@microsoft\/mgt.*)": "(\*)"/g, `"$1": "${version}"`);
-    result = result.replace(/"version": "(.*)"/g, `"version": "${version}"`);
+    let result;
```

Comment:
Do we need these changes in the setVersion script now that release-please is modifying the sonar-project file?

Reply:
I think we should just include it as other versions across the repo are also being updated in the working directory for the workflow

Reply:
The changes to the package.json files are necessary to have correct version applied when packaging the npm release. 

As I understand it the change made to the sonar-project has no lasting effect in our repo in this context.

Reply:
I'm concerned that the sonar cloud new code analysis will not use the most recent version and may wait till the next commit to main to analyze new code.

Reply:
Are you saying that Sonar Cloud is looking at the version and not doing analysis if it's seen a specific version number before? Got any docs on this for me to review?

Reply:
@gavinbarron sure, [here](https://docs.sonarsource.com/sonarcloud/improving/new-code-definition/#new-code-definition-options)

Reply:
Awesome! Thank you for that link.

In terms of PR analysis the new code version number doesn't matter at all and it's just looking at the changes that are in that PR.

So, the new code analysis is performed after changes to the main branch and looks at changes since the first time SonarCloud saw that specific version. 
Here's the workflow as I understand it:

Developers raise PRs to main (PR Analysis occurs)
PR is merged to main.
SonarCube analysis on main occurs looking at changes since the specified version (we need this step added to the workflow that triggers on pushes to main)
Release Please action triggers and might create/update a PR to main which would increment the version and changelog.

Given this we should remove the code that touches the sonarcloud version from setVersion and make changing that the responsibility of release please alone. That way the only way that a new version number is put into the main branch is via the release PR.

Reply:
I've removed the sonarCloud code from setVersion. Our Build & Test workflow triggers on pushes to main, and it included the sonar-cube analysis so I think we're covered on that.

Reply:
Nice, I wasn't aware that you'd added the push trigger to pr.yaml